testcases/InstallUpstreamKernel.py

```diff
@@ -133,18 +133,22 @@ def is_url(path):
                     cmdline += " %s" % self.append_kernel_cmdline
                 kern_rel_str = con.run_command(
                     "cat %s/include/config/kernel.release" % linux_path)[-1]
-                initrd_file = con.run_command(
-                    "ls -l /boot/initr*-%s.img" % kern_rel_str)[-1].split(" ")[-1]
-                kexec_cmdline = "kexec --initrd %s --command-line=\"%s\" /boot/vmlinuz-%s -l" % (
+                try:
+                    initrd_file = con.run_command(
+                        "ls -l /boot/initr*-%s.img" % kern_rel_str)[-1].split(" ")[-1]
+                except Exception:
+                    initrd_file = con.run_command(
+                        "ls -l /boot/initr*-%s" % kern_rel_str)[-1].split(" ")[-1]
+                kexec_cmdline = "kexec --initrd %s --command-line=\"%s\" /boot/vmlinu*-%s -l" % (
```

Comment:
As mentioned there are 2 recommended options
1) Use exec without -l option avoiding use of another follow-up kexec command
2) use kexec -l but instead of kexec -e opt for reboot.

I have coded the second option.

Reply:
ah got it, thanks